components/brave_wallet/browser/eth_tx_manager.cc

```diff
@@ -687,6 +687,61 @@ void EthTxManager::ContinueMakeERC721TransferFromData(
   std::move(callback).Run(true, data_decoded);
 }
 
+void EthTxManager::MakeERC1155TransferFromData(
+    const std::string& from,
+    const std::string& to,
+    const std::string& token_id,
+    const std::string& value,
+    const std::string& contract_address,
+    MakeERC1155TransferFromDataCallback callback) {
+  uint256_t token_id_uint = 0;
+  if (!HexValueToUint256(token_id, &token_id_uint)) {
+    VLOG(1) << __FUNCTION__ << ": Could not convert token_id";
+    std::move(callback).Run(false, std::vector<uint8_t>());
+    return;
+  }
+
+  uint256_t value_uint = 0;
+  if (!HexValueToUint256(value, &value_uint) || (value_uint == 0)) {
+    VLOG(1) << __FUNCTION__ << ": Could not convert value";
+    std::move(callback).Run(false, std::vector<uint8_t>());
+    return;
+  }
+
+  // Check if safeTransferFrom is supported first.
+  json_rpc_service_->GetSupportsInterface(
+      contract_address, kERC1155InterfaceId,
+      base::BindOnce(&EthTxManager::ContinueMakeERC1155TransferFromData,
+                     weak_factory_.GetWeakPtr(), from, to, token_id_uint,
+                     value_uint, std::move(callback)));
```

Comment:
Isn't this method always supported for ERC1155?
Why do we need to do GetSupportsInterface here?
The result boolean value seems not used in `ContinueMakeERC1155TransferFromData` either.

Reply:
I followed the implementation for MakeERC721TransferFromData which makes a supports interface call here https://github.com/brave/brave-core/blob/6fe6b6b084583cad02c44cc0e9c4ec30c804b0c5/components/brave_wallet/browser/eth_tx_manager.cc#L655  SafeTransferFrom is also a part of ERC721, so didn't see why it would apply for the 721 case but not 1155.  I did see that we do not make the supportsInterface call for ERC20s though, so I'm curious how we reconcile that.

Can we assume the caller of this function knows ahead of time knows the standard of the token?  If so, how is that determined?  If that's the case, we can omit the call here.

> The result boolean value seems not used in ContinueMakeERC1155TransferFromData either.
This is a programming oversight.  I will add or remove depending on your answers to the above.

Reply:
Replied in the DM, ERC1155 don't need this, ERC721 need this because there were some contracts implementing earlier version of ERC721 which only supports transferFrom.